impala/hiveserver2.py

```diff
@@ -1026,20 +1027,47 @@ def _execute(self, func_name, request):
             except socket.error:
                 log.exception('Failed to open transport (tries_left=%s)',
                               tries_left)
+                last_http_exception = None
             except TTransportException:
                 log.exception('Failed to open transport (tries_left=%s)',
                               tries_left)
+                last_http_exception = None
+            except HttpError as h:
+                if not retry_on_http_error:
+                    log.debug('Caught HttpError %s %s in %s which is not retryable',
+                              h, str(h.body or ''), func_name)
+                    raise
+                last_http_exception = h
+                if tries_left > 0:
+                    retry_secs = None
```

Comment:
Should we add a default sleep interval such as 1 sec and use exponential back off and progressively increase the wait interval for every retry attempt? If a failure occurred then retrying instantly would most probably result in failure again.
For the case where the http header has the retry after header we could follow the sleep duration from the header.

Reply:
I'm adding a 1 second sleep